Sources/NIOCertificateReloading/TimedCertificateReloader.swift

```diff
@@ -0,0 +1,421 @@
+        }
+    }
+
+    private struct CertificateKeyPair {
+        var certificate: NIOSSLCertificateSource
+        var privateKey: NIOSSLPrivateKeySource
+    }
+
+    private let refreshInterval: Duration
+    private let certificateDescription: CertificateDescription
+    private let privateKeyDescription: PrivateKeyDescription
+    private let state: NIOLockedValueBox<CertificateKeyPair?>
+    private let logger: Logger?
+
+    /// A `NIOSSLContextConfigurationOverride` that will be used as part of the NIO application's TLS configuration.
+    /// Its certificate and private key will be kept up-to-date via the reload mechanism the ``TimedCertificateReloader``
+    /// implementation provides.
+    public var sslContextConfigurationOverride: NIOSSLContextConfigurationOverride {
+        get {
+            var override = NIOSSLContextConfigurationOverride()
+            guard let certificateKeyPair = self.state.withLockedValue({ $0 }) else {
+                return override
+            }
+            override.certificateChain = [certificateKeyPair.certificate]
+            override.privateKey = certificateKeyPair.privateKey
+            return override
+        }
+    }
+
+    /// Initialize a new ``TimedCertificateReloader``.
+    /// - Parameters:
+    ///   - refreshInterval: The interval at which attempts to update the certificate and private key should be made.
+    ///   - certificateDescription: A ``CertificateDescription``.
+    ///   - privateKeyDescription: A ``PrivateKeyDescription``.
+    public init(
+        refreshInterval: TimeAmount,
+        certificateDescription: CertificateDescription,
+        privateKeyDescription: PrivateKeyDescription,
+        logger: Logger? = nil
+    ) {
+        self.init(
+            refreshInterval: Duration(refreshInterval),
+            certificateDescription: certificateDescription,
+            privateKeyDescription: privateKeyDescription,
+            logger: logger
+        )
+    }
+
+    /// Attempt to initialize a new ``TimedCertificateReloader``, but throw if the given certificate and private keys cannot be
+    /// loaded.
+    /// - Parameters:
+    ///   - refreshInterval: The interval at which attempts to update the certificate and private key should be made.
+    ///   - validatingCertificateDescription: A ``CertificateDescription``.
+    ///   - validatingPrivateKeyDescription: A ``PrivateKeyDescription``.
+    /// - Throws: If the certificate or private key cannot be loaded.
+    public init(
+        refreshInterval: TimeAmount,
+        validatingCertificateDescription: CertificateDescription,
+        validatingPrivateKeyDescription: PrivateKeyDescription,
+        logger: Logger? = nil
+    ) throws {
+        try self.init(
+            refreshInterval: Duration(refreshInterval),
+            validatingCertificateDescription: validatingCertificateDescription,
+            validatingPrivateKeyDescription: validatingPrivateKeyDescription,
+            logger: nil
```

Comment:
```suggestion
            logger: logger
```

Or was it intentional?

Reply:
Nope, copy pasta